src/unix/core.c

```diff
@@ -1041,6 +1041,20 @@ int uv_getrusage(uv_rusage_t* rusage) {
 }
 
 
+int uv_getrusage(uv_rusage_t* rusage) {
+  return uv__getrusage(RUSAGE_SELF, rusage);
+}
+
+
+int uv_getrusage_thread(uv_rusage_t* rusage) {
+#ifndef RUSAGE_THREAD
+  return UV_ENOTSUP;
+#else
+  return uv__getrusage(RUSAGE_THREAD, rusage);
+#endif
+}
+
```

Comment:
For `macOS`, what about using `thread_info()` as it's being done in the [node.js](https://github.com/nodejs/node/pull/56467/) [PR](https://github.com/nodejs/node/blob/1fdd439323e9c72aa0e7e0e7eed5ab03e71252aa/src/node_process_methods.cc#L175-L193)?

Reply:
That works too if you use `THREAD_EXTENDED_INFO` because then `thread_info()` returns the exact same information as `proc_pidinfo(PROC_PIDTHREADID64INFO)`.

Reply:
macOS is now supported!